Add Keycloak gatekeeper chart

gabibbo97 opened the pull request:
Keycloak gatekeeper version 4.6.0.Final

Signed-off-by: Giacomo Longo <gabibbo97@gmail.com>

#### What this PR does / why we need it:

This adds keycloak gatekeeper authenticating proxy

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [X] Variables are documented in the README.md

gabibbo97 committed:
Add Keycloak Gatekeeper chart

Signed-off-by: Giacomo Longo <gabibbo97@gmail.com>

axdotl commented on stable/keycloak-gatekeeper/values.yaml:
```diff
@@ -0,0 +1,87 @@
+affinity: {}
+
+# URL for OpenID autoconfiguration
+# On Keycloak <server>/auth/realm/<realm_name>
```
Has to be `realms` instead of `realm`
`<server>/auth/realms/<realm_name>`

gabibbo97 replied:
Changed with the latest revision fb8f3f1

gabibbo97 committed:
[stable/keycloak-gatekeeper] Add RBAC support

Signed-off-by: Giacomo Longo <gabibbo97@gmail.com>

gabibbo97 committed:
[stable/keycloak-gatekeeper] Allow to indicate encryption key

Signed-off-by: Giacomo Longo <gabibbo97@gmail.com>

gabibbo97 committed:
[stable/keycloak-gatekeeper] The service account should be used anyway

Following https://github.com/helm/helm/blob/master/docs/chart_best_practices/rbac.md

The service account should still be used from its template

Signed-off-by: Giacomo Longo <gabibbo97@gmail.com>
